### DISCUSSION TITLE
Redirect to page after login?

### DISCUSSION THREAD

**Discussion by stojankukrika** [filamentphp/filament#14280] [Help] [answered]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

Is there any option to redirect the user to a specific page after login(every time) for one of the roles? 
We have three roles: admin, creator, and buyer, and we want the if buyer to login or register go to route with name 'listing'. Is that possible?

**Comment by leandrocfe** [comment, marked as answer]
Create a custom class (app/Http/Responses/CustomLoginResponse.php)

```php
<?php

namespace App\Http\Responses;

use Filament\Http\Responses\Auth\Contracts\LoginResponse;
use Illuminate\Http\RedirectResponse;
use Livewire\Features\SupportRedirects\Redirector;

class CustomLoginResponse implements LoginResponse
{
    public function toResponse($request): RedirectResponse|Redirector
    {
        $user = auth()->user();

        return redirect()->intended(match (true) {
            default => filament()->getUrl(),
            $user->hasRole('buyer') => route('listing')
        });
    }
}
```

Then, add in your PanelProvider
```php
use App\Http\Responses\CustomLoginResponse;
use Filament\Http\Responses\Auth\LoginResponse;
...

$panel
->bootUsing(function () {
    app()->bind(LoginResponse::class, CustomLoginResponse::class);
})
...

```

You can use the same approach to customize the `RegistrationResponse`

**Reply by stojankukrika** [reply]
Thank you a lot @leandrocfe I didn't find that in the documentation.
